dandischema/digests/dandietag.py

```diff
@@ -0,0 +1,202 @@
+# Derived from <https://github.com/girder/django-s3-file-field/blob/master/
+# s3_file_field/_multipart.py>, copyright Kitware, Inc. <kitware@kitware.com>
+# under the Apache 2.0 license
+
+from dataclasses import dataclass
+from hashlib import md5
+import math
+import os
+from typing import Iterator, List, NamedTuple, Optional, Union
+
+
+def mb(bytes_size: int) -> int:
+    return bytes_size * 2 ** 20
+
+
+def gb(bytes_size: int) -> int:
+    return bytes_size * 2 ** 30
+
+
+def tb(bytes_size: int) -> int:
+    return bytes_size * 2 ** 40
+
+
+class Part(NamedTuple):
+    number: int
+    offset: int
+    size: int
+
+
+@dataclass
+class PartGenerator:
+    part_qty: int
+    initial_part_size: int
+    final_part_size: int
+
+    # S3 multipart limits: https://docs.aws.amazon.com/AmazonS3/latest/dev/qfacts.html
+    # 10k is the maximum number of allowed parts allowed by S3
+    MAX_PARTS = 10_000
+    # 5MB is the minimum part size allowed by S3
+    MIN_PART_SIZE = mb(5)
+    # 5GB is the maximum part size allowed by S3
+    MAX_PART_SIZE = gb(5)
+
+    @classmethod
+    def for_file_size(cls, file_size: int) -> "PartGenerator":
+        """Method to calculate sequential part sizes given a file size"""
+        if file_size == 0:
+            return cls(0, 0, 0)
+
+        part_size = mb(64)
+
+        if file_size > tb(5):
```

Comment:
just an fyi that on amazon this is terabytes not tebibytes.

Reply:
having said that, it's possible that amazon's info is not consistent across services. but they are generally clear about TB/TiB nomenclature.